Sources/System/Internals/WindowsSyscallAdapters.swift

```diff
@@ -35,15 +35,23 @@ internal func open(
     bInheritHandle: decodedFlags.bInheritHandle
   )
 
-  let hFile = CreateFileW(path,
-                          decodedFlags.dwDesiredAccess,
-                          DWORD(FILE_SHARE_DELETE
-                                  | FILE_SHARE_READ
-                                  | FILE_SHARE_WRITE),
-                          &saAttrs,
-                          decodedFlags.dwCreationDisposition,
-                          decodedFlags.dwFlagsAndAttributes,
-                          nil)
+  let hFile: HANDLE
+  do {
+    hFile = try path.withCanonicalPathRepresentation({ path in
+      CreateFileW(path,
+                  decodedFlags.dwDesiredAccess,
+                  DWORD(FILE_SHARE_DELETE
+                        | FILE_SHARE_READ
+                        | FILE_SHARE_WRITE),
+                  &saAttrs,
+                  decodedFlags.dwCreationDisposition,
+                  decodedFlags.dwFlagsAndAttributes,
+                  nil)
+    })
+  } catch {
+    ucrt._set_errno(_mapWindowsErrorToErrno(GetLastError()))
```

Comment:
I do not think it is right to set the errno here. (Or anywhere in swift-system.) 

These _internal_ functions should report system errors (or HRESULTs) by throwing the actual error. (As a typed error, if our toolchain version is already high enough.)

Reply:
These functions are already calling _set_errno now, so I think that's something that should come in a future enhancement, not entangled with this change. (plus they are non-throwing)

Reply:
why not change this to a

```swift
guard let hFile = try? ... else {
  ucrt._set_errno(...)
  return -1
}
```

Reply:
Done.